lib/src/test/kotlin/integration/account/AccountTest.kt

```diff
@@ -0,0 +1,262 @@
+            signature = declareTransactionPayload.signature,
+            version = declareTransactionPayload.version,
+        )
+
+        val feeEstimateRequest = provider.getEstimateFee(listOf(signedTransaction.toPayload()), BlockTag.LATEST)
+
+        assertNotNull(feeEstimateRequest)
+        val feeEstimate = feeEstimateRequest.send().first()
+        assertNotNull(feeEstimate)
+        assertNotEquals(Felt(0), feeEstimate.gasConsumed)
+        assertNotEquals(Felt(0), feeEstimate.gasPrice)
+        assertNotEquals(Felt(0), feeEstimate.overallFee)
+    }
+
+    @ParameterizedTest
+    @MethodSource("getAccounts")
+    fun `sign and send declare v1 transaction`(accountAndProvider: AccountAndProvider) {
+        assumeTrue(ConfigUtils.isTestEnabled(requiresGas = true), "Tests that require gas are skipped in the current environment. Enable with '-PenableGasTests=true'")
+
+        val (account, provider) = accountAndProvider
+        // assumeFalse(provider is JsonRpcProvider)
+        // Note to future developers experiencing failures in this test.
+        // Sometimes the test fails with "A transaction with the same hash already exists in the mempool"
+        // This error can be caused by RPC node not having access to pending transactions and therefore nonce not getting updated.
+
+        val contractCode = Path.of("src/test/resources/compiled_v0/providerTest.json").readText()
+        val contractDefinition = Cairo0ContractDefinition(contractCode)
+        val nonce = account.getNonce().send()
+
+        // Note to future developers experiencing failures in this test.
+        // 1. Compiled contract format sometimes changes, this causes changes in the class hash.
+        // If this test starts randomly falling, try recalculating class hash.
+        // 2. If it fails on CI, make sure to delete the compiled contracts before running this test.
+        // Chances are, the contract was compiled with a different compiler version.
+
+        val classHash = Felt.fromHex("0x661efb55f8bcf34ad1596936b631e6b581bfa246b99ff3f9f2d9b8fa4ff5962")
+        val declareTransactionPayload = account.signDeclare(
+            contractDefinition,
+            classHash,
+            ExecutionParams(nonce, Felt(1000000000000000L)),
+        )
+
+        val request = provider.declareContract(declareTransactionPayload)
+        val result = request.send()
+
+        Thread.sleep(10000)
+
+        val receipt = provider.getTransactionReceipt(result.transactionHash).send()
+
+        assertNotNull(result)
+        assertNotNull(receipt)
+        assertTrue(receipt.isAccepted)
+    }
+
+    @Disabled
+    // TODO: Fails with "Compiled versions older than 1.1.0 or newer than 1.3.0 are not supported. Got version 1.0.0."
+    // Can be reenabled after compiler version bump (awaiting PR #285 merge)
+    @ParameterizedTest
+    @MethodSource("getAccounts")
+    fun `sign and send declare v2 transaction`(accountAndProvider: AccountAndProvider) {
```

Comment:
Let's add more tests - ideally we should test the whole flow of declaring, deploying and invoking/calling contracts.

Reply:
Of course this does not have to be done in this PR, can be added later.
Said flow should, however, be manually tested by you at least once before merging this PR.

Reply:
I understand, that an issue is created for this? 🙏